### PR TITLE
Stats Revamp: Fix Followers Details Screen Followers Total

### PR DESCRIPTION
**Pull request by sla8c**
This PR adds more stability for the top Followers Total card on the 2nd level Stats Insights screen.  During testing it was noted that Followers Total could be showing 0 on the top card even though there are followers.

I hadn't appreciated that StatsInsightsStore has 2 methods of requesting followers for example:
 * dotComFollowers
 * allDotComFollowers

For the purposes of the Followers Total card summary, I've changed the call to use the dotComFollowers which returns quicker and is consistent with the Top Level Followers Total insights card

Ref #18605

Screenshot below of the problem.  Note that this inconsistent Followers Total count does not always manifest (I could only get it to happen 1 out of 20 times)

image

Expected result:
image

To test:
	1. enable both of the new stats feature flags statsNewAppearance and statsNewInsights via AppConfiguration.showsStatsRevampV2
	2. From top level Stats Insights screen, Find the Followers Total card (or add if needed) and take note of the count.  Then tap "View More" on the Followers Total card. You should be taken to the 2nd Level Insights Details and should see Followers Total card at the top
	3. The count of the Followers Total card on the 2nd level should match the count of the Followers Total card on the top level
Switch sites and continue to test this for other sites

## Regression Notes
1. Potential unintended areas of impact
Stats revamp v2

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by sla8c** [comment]
I've set this to @guarani for review and cc @frosty for fyi.  Thanks

File: WordPress/Classes/Stores/StatsInsightsStore.swift [hunk 1]
```diff
@@ -1087,6 +1087,19 @@ extension StatsInsightsStore {
...
+        switch (followersTotalsStatus, state.dotComFollowersStatus, state.emailFollowersStatus) {
+        case (let a, let b, let c) where a == .loading || b == .loading || c == .loading:
+            return .loading
+        case (let a, let b, let c) where a == .error || b == .error || c == .error:
+            return .error
+        case (.success, .success, .success):
+            return .success
+        default:
+            return .idle
+        }
```
Comment (guarani) [inline WordPress/Classes/Stores/StatsInsightsStore.swift]: @sla8c, the `a`, `b`, and `c` temporary variables and advanced switch statement make this a bit hard for me to follow. Would something like this work and be simpler? 

```suggestion
        let statuses = [followersTotalsStatus, state.dotComFollowersStatus, state.emailFollowersStatus]
        if statuses.contains(.loading) {
            return .loading
        } else if statuses.contains(.error) {
            return .error
        } else if statuses.allSatisfy({ $0 == .success }) {
            return .success
        } else {
            return .idle
        }
```

**Review by guarani — COMMENTED** [review]
Is the difference between `dotComFollowers` and `allDotComFollowers` that the former returns the total follower count + only the top followers while the latter return the total follower count + all followers?

I wasn't able to reproduce the issue that this PR addresses, but I noticed you said it was only reproducible 1 in 20 times. Does it look to be an API issue, an app issue, or you're not sure?

Since this PR targets next release, not the current release, I'll hold off giving a review to better understand the above questions.

File: WordPress/Classes/Stores/StatsInsightsStore.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/Stores/StatsInsightsStore.swift]: @Guarani @staskus looking at this with a fresh set of eyes, I have removed my function here however I added a unit test to test the existing function but did not refactor it yet (I'll do so in another PR)

Comment (staskus) [reply WordPress/Classes/Stores/StatsInsightsStore.swift]: Could you expand a bit more about what @guarani asked before: the differences between 2 methods and do you know why exactly the issue happens?

**Comment by staskus** [comment]
I see that "before" screenshot shows the date switcher while "expected" screenshot doesn't show the date switcher on top.

Is it also related to this change? What is the expectation?

**Comment by staskus** [comment]
✅ Tested. Followers counts match at both levels. The count also matches the web.

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift [hunk 2]
```diff
@@ -527,8 +529,8 @@ class SiteStatsInsightsDetailsViewModel: Observable {
                 forceRefresh: forceRefresh))
     }
 
-    func refreshFollowers() {
-        ActionDispatcher.dispatch(InsightAction.refreshFollowers)
+    func refreshFollowers(forceRefresh: Bool = true) {
+        ActionDispatcher.dispatch(InsightAction.refreshInsights(forceRefresh: forceRefresh))
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift:533]: Do I understand correctly that all the possible insight data is refreshed when a user opens "Total Followers" view?

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift [hunk 3]
```diff
@@ -634,7 +636,7 @@ private extension SiteStatsInsightsDetailsViewModel {
     func queryForInsightStatSection(_ statSection: StatSection) -> InsightQuery? {
         switch statSection {
         case .insightsFollowersWordPress, .insightsFollowersEmail, .insightsFollowerTotals:
-            return .allFollowers
+            return .insights // use .insights here which is same as top level insights screen
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift:639]: Is a change like this replicable for other types of sections (like comments)? Or are they specific and this change is only relevant to followers?

I think this question boils down to the difference between `getAllDotComFollowers` vs `getDotComFollowers`.

File: WordPress/Classes/Stores/StatsInsightsStore.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/Stores/StatsInsightsStore.swift]: @staskus the difference was that the new function I added was redundant and I could just re-use `followersTotalsStatus`

**Comment by sla8c** [comment]
> I see that "before" screenshot shows the date switcher while "expected" screenshot doesn't show the date switcher on top.
> 
> Is it also related to this change? What is the expectation?

I think that's as a result of stats revamp being in flux at that time the PR was created months ago.  The expectation is for the date switcher to not be there (and its not there now).  It was determined that the date switcher was not needed for 'Totals'

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift (hunk 2 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift:533]: That's correct - there is room for optimisation on all the store calls I think

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift (hunk 3 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift:639]: .allFollowers returns `getAllDotComFollowers` and for insights, we don't need to make a large-ish network call for AllDotComFollowers we just need the top 10 per screen designs which is provided by the fetchInsights call ie Insights is more of a summary.

Comment (staskus) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift:639]: Okay got it now. Thanks for the explanation 👍

**Review by staskus — APPROVED** [review]
Thank you for the answers about the code. The testing also produced expected results 👍

**Comment by staskus** [comment]
>  think that's as a result of stats revamp being in flux at that time the PR was created months ago. The expectation is for the date switcher to not be there (and its not there now). It was determined that the date switcher was not needed for 'Totals'

@sla8c 

I noticed that there're 2 different followers views:
- Insights -> Followers (Contains data picker)
- Insights -> Total Followers (Doesn't contain data picker)

However, I think both **shouldn't** contain a data picker.

Created an issue https://github.com/wordpress-mobile/WordPress-iOS/issues/19763

**Pull request merged**